### PR TITLE
US017 edit band

### PR DESCRIPTION
**Pull request by JamesMcKee-Kainos**

### CODE REVIEW

File: src/main/java/org/kainos/ea/dao/TrainingCourseDao.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+        ResultSet rs = st.executeQuery("SELECT training_courses.id, training_courses.name, training_courses.description FROM training_courses " +
+                "INNER JOIN band_training_courses ON training_courses.id = band_training_courses.training_course_id " +
+                "WHERE band_training_courses.band_id = " + id + ";");
+
+        List<TrainingCourse> trainingCourseList = new ArrayList<>();
+
+        while (rs.next()) {
+            TrainingCourse trainingCourse = new TrainingCourse(
+                    rs.getInt("training_courses.id"),
```
[the remark this answers is not in the document]
Comment (JamesMcKee-Kainos) [reply src/main/java/org/kainos/ea/dao/TrainingCourseDao.java:68]: This is selecting by the band ID - so we still need to get the training course ID

File: src/main/java/org/kainos/ea/controller/CompetencyController.java [hunk 2]
```diff
@@ -28,4 +28,16 @@ public Response getAllCompetencies() {
             return Response.serverError().build();
         }
     }
+
+    @GET
+    @Path("/competencies/band/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getBandCompetencies(@PathParam("id") int id) {
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/controller/CompetencyController.java:35]: could this be returned by the GET bands by ID endpoint?

File: src/main/java/org/kainos/ea/controller/TrainingCourseController.java [hunk 3]
```diff
@@ -29,4 +27,16 @@ public Response getAllTrainingCourses() {
             return Response.serverError().build();
         }
     }
+
+    @GET
+    @Path("/training-courses/band/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getTrainingForBand(@PathParam("id") int id) {
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/controller/TrainingCourseController.java:34]: could this be returned by the GET bands by ID endpoint?

File: src/test/java/integration/controller/CompetencyControllerIntegrationTest.java [hunk 4]
```diff
@@ -28,4 +28,13 @@ void getAllCompetencies_shouldReturnList() {
 
         Assertions.assertTrue(response.size() > 0);
     }
+
+    @Test
+    void getBandCompetencies_shouldReturnList() {
```
Comment (shaunganley) [inline src/test/java/integration/controller/CompetencyControllerIntegrationTest.java:33]: needs invalid ID test

File: src/test/java/integration/controller/TrainingCourseControllerIntegrationTest.java [hunk 5]
```diff
@@ -27,4 +27,13 @@ void getAllTrainingCourses_shouldReturnList() {
 
         Assertions.assertTrue(response.size() > 0);
     }
+
+    @Test
+    void getTrainingForBand_shouldReturnList() {
```
Comment (shaunganley) [inline src/test/java/integration/controller/TrainingCourseControllerIntegrationTest.java:32]: needs invalid ID test

**Review by shaunganley — CHANGES_REQUESTED** [review]
few small comments

**Review by joshuasy10 — APPROVED** [review]

**Review by shaunganley — APPROVED** [review]

**Pull request merged**